pkg/controller/machinepool/awsactuator.go

```diff
@@ -317,13 +335,6 @@ func (a *AWSActuator) updateProviderConfig(machineSet *machineapi.MachineSet, in
 		}
 	}
 
-	providerConfig.SecurityGroups = []machineapi.AWSResourceReference{{
```

Comment:
NTR: This was overwriting the filter generated by the installer's provider code. AFAICS, until this feature, it was overwriting it with the exact same value, generated the exact same way. Please scrutinize for any discrepancy I may have missed that would make removing this Bad™.